DEVELOPERS.md

```diff
@@ -0,0 +1,421 @@
+# Developers Guide
+
+This guide is to assist developers who wish to modify one or more of the model components of ACCESS-OM2 and compile the modified code. Typically there is a development cycle, where the code is modified, compiled, tested and further modified based on testing.
+
+In particular this guide steps through the process to develop the ACCESS-OM2 model using [`spack`](spack.io), the build from source package manager that is use to build and deploy the model.
+
+It is assumed the modifications and compilation will be done on the [NCI HPC system `gadi`](https://opus.nci.org.au/display/Help/Gadi+User+Guide).
+
+Much of this adapted from the [spack Developer Workflows Tutorial](https://spack-tutorial.readthedocs.io/en/latest/tutorial_developer_workflows.html).
+
+## Initial set-up 
+
+The steps in this section need only be done once, as long as the versions of the three repositories are sufficient.
+
+It is necessary to make a local copy of `spack`, the configuration suitable for `gadi` (`spack-config`) and a package repository (`spack-packages`) which defines a number of the packages used by ACCESS-OM2.
+
+Adapted from [instructions on the forum](https://forum.access-hive.org.au/t/how-to-build-access-om2-on-gadi/1545)
+
+```bash
+git clone -c feature.manyFiles=true https://github.com/ACCESS-NRI/spack.git --branch releases/v0.21 --single-branch --depth=1
+git clone https://github.com/ACCESS-NRI/spack-packages.git --branch main
+git clone https://github.com/ACCESS-NRI/spack-config.git --branch main
+```
+
+Note: `spack` is cloned from the ACCESS-NRI fork as on occasion there are some fixes that are back-ported by ACCESS-NRI.
```

Comment:
> there are some fixes that are back-ported by ACCESS-NRI

Doesn't this mean there are fixes done on the official spack repo and then you get them on the ACCESS-NRI fork? If so, how is it better to work from the fork than from the official repo directly?

Reply:
No it means we've back-ported some fixes that the official `spack` version has not. Typically it is changes to the builtin packages.